### PR TITLE
adds tests for endpoint

### PR DESCRIPTION
**Pull request by FelixPriori**
# In this PR
## tests for endpoints
* `/badges`
* `/classes`
* `/users`
* `/quests`

### CODE REVIEW

File: back-end/package.json [hunk 1]
```diff
@@ -5,9 +5,9 @@
   "scripts": {
     "server": "node server.js",
     "dev": "nodemon server",
-    "lint": "eslint -c .eslintrc.json --ext .js .",
+    "lint": "eslint -c .eslintrc.json --ext .js . --fix",
```
Comment (cawel) [inline back-end/package.json:8]: Adding back that `--fix` option breaks the pre-commit git hook. See [this PR setting up husky](#13) for details.

If you want to fix the ESLint issues, you could use `npm run link:fix`

File: back-end/tests/endpoints.test.js [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+
+describe('quests', () => {
+
+  it('should return a quest based on the id = 1 ', async done => {
+    const response = await request.get('/quests/1');
+    expect(response.status).toBe(200);
+    expect(response.body.length).toBe(1);
+    expect(response.body[0]).toMatchObject({
+      id: 1,
+      name: 'Can\'t figure out this math problem.',
```
Comment (cawel) [inline back-end/tests/endpoints.test.js:162]: Did you know you could use double quotes in cases like these?
```js
name: "Can't figure out this math problem."
```
That way you don't need the backslash to escape the apostrophe.

File: back-end/package.json (hunk 1 above)
Comment (FelixPriori) [reply back-end/package.json:8]: Sorry about that! I'll fix it tmrw first thing.

File: back-end/tests/endpoints.test.js (hunk 2 above)
Comment (FelixPriori) [reply back-end/tests/endpoints.test.js:162]: I did know. But why I didn't do it like that: I have no idea. ¯\_(ツ)_/¯

Comment (cawel) [reply back-end/tests/endpoints.test.js:162]: I find it easier to read using double quotes in this case. Personal preference :)

File: back-end/tests/endpoints.test.js [hunk 3]
```diff
@@ -0,0 +1,207 @@
+/* eslint-disable camelcase */
+/* eslint-disable no-undef */
```
Comment (cawel) [inline back-end/tests/endpoints.test.js:2]: I did not have issues running the linter after removing those 2 lines. Could you give me an example of an offending line in this file?

Comment (FelixPriori) [reply back-end/tests/endpoints.test.js:2]: `eslint-disable camelcase` =>
```
      int_requirement: 1,
      criteria_type: "quest",
      class_id: 1
```
And actually `eslint-disable no-undef` used to be anything that has `expect(...)`
but not it doesn't complain anymore.

File: back-end/tests/endpoints.test.js [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+    expect(response.body[0]).toMatchObject({
+      id: 1,
+      name: "A stealthy aquaintance",
+      requirement: "Complete 1 Rogue Quest",
+      int_requirement: 1,
+      criteria_type: "quest",
+      class_id: 1
+    });
+    done();
+  });
```
Comment (cawel) [inline back-end/tests/endpoints.test.js:21]: All the tests in this file could be made more readable if they were not using the `done()` callback.

For instance:

```js
it('should return an array of objects', async () => {
    const response = await request.get('/badges');
    expect(response.status).toBe(200);
    expect(response.body.length).toBeGreaterThan(0);
    expect(response.body[0]).toMatchObject({
      id: 1,
      name: "A stealthy acquaintance",
      requirement: "Complete 1 Rogue Quest",
      int_requirement: 1,
      criteria_type: "quest",
      class_id: 1
    });
});
```

See [Jest documentation](https://jestjs.io/docs/en/asynchronous#asyncawait).

File: back-end/tests/endpoints.test.js (hunk 3 above)
Comment (cawel) [reply back-end/tests/endpoints.test.js:2]: Your `eslint-disable no-undef` issue was probably fixed when we added `jasmine` in the ESLint config.

Comment (cawel) [reply back-end/tests/endpoints.test.js:2]: And you have that camel case issue in a shell when running `npm run lint` ?

Comment (FelixPriori) [reply back-end/tests/endpoints.test.js:2]: Yes.

```
16:7  error  Identifier 'int_requirement' is not in camel case    camelcase
   17:7  error  Identifier 'criteria_type' is not in camel case      camelcase
   18:7  error  Identifier 'class_id' is not in camel case           camelcase
   30:7  error  Identifier 'int_requirement' is not in camel case    camelcase
   31:7  error  Identifier 'criteria_type' is not in camel case      camelcase
   32:7  error  Identifier 'class_id' is not in camel case           camelcase
   69:7  error  Identifier 'int_requirement' is not in camel case    camelcase
   70:7  error  Identifier 'criteria_type' is not in camel case      camelcase
   71:7  error  Identifier 'class_id' is not in camel case           camelcase
  105:7  error  Identifier 'first_name' is not in camel case         camelcase
  106:7  error  Identifier 'last_name' is not in camel case          camelcase
  123:7  error  Identifier 'first_name' is not in camel case         camelcase
  124:7  error  Identifier 'last_name' is not in camel case          camelcase
  141:7  error  Identifier 'first_name' is not in camel case         camelcase
  142:7  error  Identifier 'last_name' is not in camel case          camelcase
(...)
```

Comment (cawel) [reply back-end/tests/endpoints.test.js:2]: Hum. I wonder why we don't have the some output 🤔

**Pull request merged**